### PR TITLE
Ensure the command are lower case for RedisLoadBalancerContext

### PR DESCRIPTION
**Pull request by HenryYYang**
Description: Found a similar issue with #8530 in the RedisLoadBalancerContext.  The redis command needs to be lower case when we determine if it's a read command.
Risk Level: Low
Testing: Add unit test
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by HenryYYang** [comment]
PTAL @FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg @msukalski

File: source/extensions/clusters/redis/redis_cluster_lb.cc [hunk 1]
```diff
@@ -171,9 +171,15 @@ bool isReadRequest(const NetworkFilters::Common::Redis::RespValue& request) {
...
     return false;
   }
-  return NetworkFilters::Common::Redis::SupportedCommands::isReadCommand(first.asString());
+  std::string to_lower_string(first.asString());
+  toLowerTable().toLowerCase(to_lower_string);
+  return NetworkFilters::Common::Redis::SupportedCommands::isReadCommand(to_lower_string);
+}
+
+const ToLowerTable& RedisLoadBalancerContextImpl::toLowerTable() {
+  static auto* table = new ToLowerTable();
```
Comment (junr03) [inline source/extensions/clusters/redis/redis_cluster_lb.cc]: should we make this a static member on the class itself?

Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply source/extensions/clusters/redis/redis_cluster_lb.cc]: I did that for the Redis command stats, following the , but the HTTP codec lays forth the pattern Henry is using.

If it's ASCII, which a Redis command should be, you could always use `absl::AsciiStrToLower`.

Comment (HenryYYang) [reply source/extensions/clusters/redis/redis_cluster_lb.cc]: This supports lazy initialization, the static member on the class will be initialized even if there's no calls to the toLowerTable() function.  With the local static variable, it only initialized when it's used.

**Pull request merged**